src/DependencyInjection/CompilerPass/MakeCommandRegistrationPass.php

```diff
@@ -32,13 +33,12 @@ public function process(ContainerBuilder $container)
                 throw new InvalidArgumentException(sprintf('Service "%s" must implement interface "%s".', $id, MakerInterface::class));
             }
 
-            $container->register(
-                sprintf('maker.auto_command.%s', Str::asTwigVariable($class::getCommandName())),
-                MakerCommand::class
-            )->setArguments([
-                new Reference($id),
-                new Reference('maker.file_manager'),
-            ])->addTag('console.command', ['command' => $class::getCommandName()]);
+            $commandDefinition = new ChildDefinition('maker.auto_command.abstract');
+            $commandDefinition->setClass(MakerCommand::class);
```

Comment:
It is needed - actually `class` not inherits to child services. However it may be bug. [Documentation](https://symfony.com/doc/current/service_container/parent_services.html) says:
> All attributes on the parent service are shared with the child except for `shared`, `abstract` and `tags`. These are not inherited from the parent.

Reply:
I debugged it. Really `class` inherits. It is resolved in `ResolveChildDefinitionsPass` but `AddConsoleCommandPass` calls earlier (when `class` is set to `null`).